### PR TITLE
Relicense Dolphin under GPLv2+

### PR DESCRIPTION
**Pull request by Tilka**
~~**Don't merge this yet!**~~

Right now, Dolphin has multiple license conflicts:

1. On Android, we link to libraries which are licensed under Apache License 2.0 which is incompatible with GPLv2 but compatible with GPLv3.
2. Several libraries that we always link are GPLv3 which is incompatible with GPLv2.

To resolve these issues, we are attempting to relicense from "GPLv2" to "GPLv2 or later" aka "GPLv2+". (In practice, Dolphin as a whole would only be usable under GPLv3 but parts of its source code could be reused under GPLv2.) All past and present contributors have been asked to agree. Some couldn't be reached because their email bounced, some have refused, some have additional demands, some haven't answered. New contributions are already covered by GPLv2+ since December 1st, 2014.

TODO (everyone):
- [x] Get ~~100%~~>95% agreement, either by managing to contact the remaining contributors or by dropping/rewriting their code.
- [x] Make sure all [licenses of linked code](https://github.com/Tilka/dolphin/blob/gplv2%2B/Externals/licenses.md) are compatible with GPLv3.
- ~~Implement credits in Dolphin as required by some licenses. @JMC47 suggested adding a menu entry that starts emulation of a homebrew program.~~ Postponed.
- ~~Create a contributor credits page on the website to meet demands of one contributor. For all I care, this could simply be a link to [the contributors list on GitHub](https://github.com/dolphin-emu/dolphin/graphs/contributors).~~ No copyrightable code remaining. We'll do something anyway.
- [x] Update any new references from GPLv2 to GPLv2+.

Alternatives:
- Drop Android support. This would be a shame.
- Ignore the license issue and possibly get sued or taken down from the app store.

In terms of commits, [current progress](https://docs.google.com/spreadsheets/d/1QeZYHduULTSZWUgMg_uUghIWqsCuTvvW5Z2YCfHM7oM/edit) is at 96.53%.

### CODE REVIEW

**Comment by DZCreeper** [comment]
Not a contributor but if anyone is wondering the differences between GPL2 and GPL3, I found this:

http://www.ifross.org/en/what-difference-between-gplv2-and-gplv3

**Comment by tueidj** [comment]
Why bother with this? Don't the Android libraries count as basic OS fundamentals and are no different from linking with kernel32 and friends on windows?

**Comment by Sonicadvance1** [comment]
No.

**Comment by RachelBryk** [comment]
>Create a contributor credits page on the website to meet demands of one contributor. As far as I care, this could simply be a link to the contributors list on GitHub. 

That doesn't work since anyone without a github account won't be listed.

**Comment by Tilka** [comment]
@galop1n: If you agree with this license change, please send an email to [delroth](mailto:Pierre Bourdon <delroth+relic@dolphin-emu.org) saying "I authorize my contributions to Dolphin to be relicensed under the GPLv2+ license."
Thanks! :dolphin:

**Comment by Tilka** [comment]
@RolandMunsil: If you agree with this change, please send an email to [delroth](mailto:delroth+relic@dolphin-emu.org) saying "I authorize my contributions to Dolphin to be relicensed under the GPLv2+ license." Thank you!

**Comment by Tilka** [comment]
@LinesPrower If you agree with this change, please send an email to [delroth](delroth+relic@dolphin-emu.org) saying "I authorize my contributions to Dolphin to be relicensed under the GPLv2+ license." Thank you!

**Comment by MaJoRoesch** [comment]
As an alternative to the homebrew credit system, a button could be added to the about menu to make a new window with a contributors listing, or a link in the help menu.

It just seems like it would be annoying to keep homebrew updated. Is WX so bad that making an entirely new program is easier? lol

**Comment by BhaaLseN** [comment]
The idea of using a homebrew would be to have something to run even if you just downloaded Dolphin for the first time, and kinda to show off some of its features. It was just a shot in the blue, we could just aswell include a dialog somewhere shown from the about screen, do a Star Wars scroller if you enter the Konami code, or whatever.
Besides, the homebrew would just run; wheres we'd have to write one for WX, Qt, Android and potentially also the NoGUI builds if we wanted them there for whatever reason.

**Comment by lioncash** [comment]
Can we, like, not overcomplicate this and just redirect to https://github.com/dolphin-emu/dolphin/graphs/contributors for credits. This really isn't something to bust rocks over. Yes, those without Github will not be listed, however, seeing as the majority do, I'd like to ask: "Is it worth it?".

**Comment by BhaaLseN** [comment]
Probably not. Especially since the majority (if not all of the...at least major contributors) have a GitHub account anyways. At the moment it shows 100 members, and the Excel sheet has about 150.
I'd say it comes down to including or excluding about 50 people which amount to maybe 1% of the commits.
Besides, and while this ties more into the licensing issue, we don't even know if their code by commit is still part of the code base.

File: Source/Core/Common/Hash.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
-// Copyright 2013 Dolphin Emulator Project
-// Licensed under GPLv2
+// Copyright 2008 Dolphin Emulator Project
```
Comment (heinermann) [inline Source/Core/Common/Hash.cpp]: Why are some of these being changed to 2008 while others changed to 2014?

Comment (Tilka) [inline Source/Core/Common/Hash.cpp]: The year changes are part of 5afbcedc748043c46938e9889c4c9f7fd2ed724d.

**Comment by Tilka** [comment]
OK!

**Comment by degasus** [comment]
LGTM

**Comment by BhaaLseN** [comment]
On the topic of those damn copyright year references in the file headers (inb4 running a tool every year to change every file):
> -13:29:21- [Bh44L] flacs: is this documented somewhere? like, Contributing.md?
> -13:29:34- [Bh44L] as in: "if you create a new file, put the current year in the header"
> -13:29:34- [flacs] nope
> -13:29:39- [Bh44L] maybe we should
> -13:29:51- [Bh44L] thats what created this mess in first place

Other than that, get it merged.

**Comment by delroth** [comment]
@dolphin-emu-bot allowmerge

**Comment by hrydgard** [comment]
Hooray! Finally! Great work everyone.

**Pull request merged**